samples/CoreAnimationTest/CoreAnimationTest/CALayerViewController.m

```diff
@@ -0,0 +1,226 @@
+    [_menuTVC addMenuItemView:_switchDisplayContents andTitle:@"Display Contents"];
+
+    _switchGeometryFlipped = [[UISwitch alloc] init];
+    [_switchGeometryFlipped addTarget:self action:@selector(geometryFlippedToggle) forControlEvents:UIControlEventValueChanged];
+    [_menuTVC addMenuItemView:_switchGeometryFlipped andTitle:@"Geometry Flipped"];
+
+    _switchHidden = [[UISwitch alloc] init];
+    [_switchHidden addTarget:self action:@selector(hiddenToggle) forControlEvents:UIControlEventValueChanged];
+    [_menuTVC addMenuItemView:_switchHidden andTitle:@"Hidden"];
+
+    _sliderOpacity = [[UISlider alloc] initWithFrame:CGRectMake(0.0, 0.0, 100.0, 38.0)];
+    _sliderOpacity.backgroundColor = [UIColor clearColor];
+    _sliderOpacity.minimumValue = 0.0f;
+    _sliderOpacity.maximumValue = 1.0f;
+    _sliderOpacity.continuous = YES;
+    _sliderOpacity.value = 1.0f;
+    [_sliderOpacity addTarget:self action:@selector(opacityValueChanged) forControlEvents:UIControlEventValueChanged];
+    [_menuTVC addMenuItemView:_sliderOpacity andTitle:@"Opacity"];
+
+    _sliderBackgroundColor = [[UISlider alloc] initWithFrame:CGRectMake(0.0, 0.0, 100.0, 38.0)];
+    _sliderBackgroundColor.backgroundColor = [UIColor clearColor];
+    _sliderBackgroundColor.minimumValue = 0.0f;
+    _sliderBackgroundColor.maximumValue = 360.0f;
+    _sliderBackgroundColor.continuous = YES;
+    _sliderBackgroundColor.value = 360.0f;
+    [_sliderBackgroundColor addTarget:self action:@selector(backgroundColorValueChanged) forControlEvents:UIControlEventValueChanged];
+    [_menuTVC addMenuItemView:_sliderBackgroundColor andTitle:@"Background color"];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+}
+
+- (void)displayContentsToggle {
+    _layer.contents = _switchDisplayContents.on ? _safeImage : nil;
+}
+
+- (void)geometryFlippedToggle {
+    _layer.geometryFlipped = _switchGeometryFlipped.on;
+}
+
+- (void)hiddenToggle {
+    _layer.hidden = _switchHidden.on;
+}
+
+- (void)gravityValueChanged {
+    _layer.contentsGravity = _contentsGravityValues[(int)_sliderGravity.value];
+}
+
+- (void)opacityValueChanged {
+    _layer.opacity = _sliderOpacity.value;
+}
+
+- (void)backgroundColorValueChanged {
+    CGFloat hue = _sliderBackgroundColor.value / 359.0f;
+    UIColor* color = [UIColor colorWithHue:hue saturation:hue brightness:0.4f alpha:1.0f];
+    _layer.backgroundColor = color.CGColor;
+}
+
+- (void)setupLayer {
+    _layer = [[CALayer alloc] init];
+
+    _layer.contents = _safeImage;
+    // _layer.contentsRect
+    // _layer.contentsCenter
+    // _layer.display
+    // _layer.drawInContext
+
+    _layer.contentsGravity = kCAGravityCenter;
+    // _layer.opacity
+    // _layer.hidden
+    // _layer.masksToBounds
+    // _layer.mask
+    // _layer.doubleSided
+
+    //_layer.cornerRadius = 100.0;
+    //_layer.borderWidth = 12.0;
+    //_layer.borderColor = [UIColor whiteColor].CGColor;
```

Comment:
Sure

---
In reply to: [84970301](https://github.com/Microsoft/WinObjC/pull/1204#discussion_r84970301) [](ancestors = 84970301,84962274,84809439)